framework/math/PETScUtils/petsc_utils_04_vecmanip.cc

```diff
@@ -2,54 +2,72 @@
 
 #include "chi_log.h"
 
-//###################################################################
+// ###################################################################
 /**Copies a PETSc vector to a STL vector. Only the local portion is
  * copied.*/
-void chi_math::PETScUtils::CopyVecToSTLvector(
-  Vec x, std::vector<double>& data, size_t N)
+void chi_math::PETScUtils::CopyVecToSTLvector(Vec x,
+                                              std::vector<double>& data,
+                                              size_t N,
+                                              bool resize_STL /*=true*/)
 {
-  data.clear();
-  data.resize(N,0.0);
+  if (resize_STL)
+  {
+    data.clear();
+    data.assign(N, 0.0);
+  }
+
   const double* x_ref;
-  VecGetArrayRead(x,&x_ref);
+  VecGetArrayRead(x, &x_ref);
 
-  for (size_t i=0; i<N; ++i)
+  for (size_t i = 0; i < N; ++i)
     data[i] = x_ref[i];
 
-  VecRestoreArrayRead(x,&x_ref);
+  VecRestoreArrayRead(x, &x_ref);
 }
 
-//###################################################################
+// ###################################################################
 /**Copies a PETSc vector to a STL vector. Only the local portion is
  * copied.*/
 void chi_math::PETScUtils::CopyVecToSTLvectorWithGhosts(
-  Vec x, std::vector<double>& data, size_t N)
+  Vec x, std::vector<double>& data, size_t N, bool resize_STL /*=true*/)
 {
-  data.clear();
-  data.resize(N,0.0);
+  if (resize_STL)
```

Comment:
Same comment as above.  Needs assert() or throw an error if data.size() != N.

Reply:
Got it. This one would be .size() == N